packages/layout/src/slots/Target.tsx

```diff
@@ -1,59 +1,114 @@
 import { useClassName, useId } from '@contember/react-utils'
-import { assert, dataAttribute, isNonEmptyTrimmedString } from '@contember/utilities'
+import { assert, dataAttribute, isArrayOfMembersSatisfyingFactory, isNonEmptyArray, isNonEmptyTrimmedString, isSingleWordString, satisfiesOneOfFactory, setHasOneOf } from '@contember/utilities'
+import { snakeCase } from 'change-case'
 import { memo, useLayoutEffect, useRef } from 'react'
-import slugify from 'slugify'
 import { useActiveSlotPortalsContext, useTargetsRegistryContext } from './contexts'
-import { TargetProps } from './types'
+import { SlotTargetProps } from './types'
 
 export type OwnTargetContainerProps = {
-	'data-has-own-children': boolean;
-	'data-name': string;
 	className: string;
 }
 
+const isNameString = satisfiesOneOfFactory(
+	isNonEmptyTrimmedString,
+	isSingleWordString,
+)
+
+const nonEmptyArrayOfNonEmptyStrings = satisfiesOneOfFactory(
+	isNonEmptyArray,
+	isArrayOfMembersSatisfyingFactory(isNameString),
+)
+
 /**
  * @group Layout
  */
-export const Target = memo<TargetProps>(
+export const Target = memo<SlotTargetProps>(
 	({
 		as,
+		aliases,
 		componentClassName = 'slot',
 		className: classNameProp,
-		children,
+		display,
+		fallback,
 		name,
 		...rest
 	}) => {
-		assert('name is non-empty string', name, isNonEmptyTrimmedString)
+		assert('name is non-empty string without spaces', name, isNameString)
+
+		if (aliases) {
+			assert('aliases is an empty array or an array of names', aliases, nonEmptyArrayOfNonEmptyStrings)
+		}
 
 		const ref = useRef<HTMLElement>(null)
-		const instanceId = useRef(Math.random().toString(36).substring(2, 9)).current
+		const idRef = useRef(Math.random().toString(36).substring(2, 9))
 		const { unregisterSlotTarget, registerSlotTarget } = useTargetsRegistryContext()
 		const activeSlotPortals = useActiveSlotPortalsContext()
 
 		useLayoutEffect(() => {
-			registerSlotTarget(instanceId, name, ref)
-			return () => unregisterSlotTarget(instanceId, name)
-		}, [instanceId, name, registerSlotTarget, unregisterSlotTarget])
+			const id = idRef.current
+			registerSlotTarget(id, name, ref)
+
+			return () => {
+				unregisterSlotTarget(id, name)
+			}
+		}, [name, registerSlotTarget, unregisterSlotTarget])
+
+		const registeredAliasesRef = useRef<Set<string>>(new Set())
+
+		useLayoutEffect(() => {
+			const id = idRef.current
+
+			if (aliases) {
+				const aliasesSet = new Set(aliases)
+
+				aliasesSet.forEach(name => {
+					if (!registeredAliasesRef.current.has(name)) {
+						registerSlotTarget(id, name, ref)
+						registeredAliasesRef.current.add(name)
+					}
+				})
+
+				registeredAliasesRef.current.forEach(name => {
+					if (!aliasesSet.has(name)) {
+						unregisterSlotTarget(id, name)
+						registeredAliasesRef.current.delete(name)
+					}
+				})
+			}
+
+			const _registeredAliasesRef = registeredAliasesRef
```

Comment:
this is probably not what you want. you are still referencing the same object

Reply:
you should imo move the cleanup to separate useLayoutEffect to handle component unmount and here, only handle changes in alias array

Reply:
Done